src/m365/todo/commands/list/list-remove.ts

```diff
@@ -114,10 +73,51 @@ class TodoListRemoveCommand extends GraphCommand {
       });
 
       if (result.continue) {
-        await removeList();
+        await this.removeList(args);
       }
     }
   }
+
+  private async getListId(args: CommandArgs): Promise<string> {
+    if (args.options.name) {
+      // Search list by its name
+      const requestOptions: any = {
+        url: `${this.resource}/v1.0/me/todo/lists?$filter=displayName eq '${escape(args.options.name)}'`,
+        headers: {
+          accept: "application/json;odata.metadata=none"
+        },
+        responseType: 'json'
+      };
+      const response: any = await request.get(requestOptions);
+
+      return response.value && response.value.length === 1 ? response.value[0].id : null;
+    }
+
+    return args.options.id as string;
+  }
+
+  private async removeList(args: CommandArgs): Promise<void> {
+    try {
+      const listId: string = await this.getListId(args);
+
+      if (!listId) {
```

Comment:
Let's throw in the `getListId` function instead of here.

Reply:
good catch, totally missed that one

Reply:
I think you missed updating it :)

Reply:
Wait.. you weren't talking about the Promise.resolve() that was still present 😂

Reply:
No I meant move the throw to the getListId function. That's a more logical place for it 😀

Reply:
Of course, removing any promise.resolves is also necessary